### PR TITLE
LPS-129061 Change put...Permissions endpoints to return the list of updated permissions permission

### PR DESCRIPTION
**Pull request by javierdearcos**
Requested by Pablo, I've made a test to return the list of updated permissions from the permissions endpoints, put method.

I think this add consistency with the usual PUT usage that we have in the Portal but it also changes an existent behavior of the APIS (don't know why baseline does not bump major, but I think it has to)

Creating this PR to discuss with @nhpatt and check if it worth changing an existent behavior with the benefits of consistency with other PUT operations in the portal

### CODE REVIEW

File: modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/base_resource_impl.ftl [hunk 1]
```diff
@@ -217,6 +217,8 @@ public abstract class Base${schemaName}ResourceImpl
 					PermissionUtil.checkPermission(ActionKeys.PERMISSIONS, groupLocalService, resourceName, resourceId, getPermissionCheckerGroupId(${schemaVarName}Id));
 
 					resourcePermissionLocalService.updateResourcePermissions(contextCompany.getCompanyId(), 0, resourceName, String.valueOf(resourceId), ModelPermissionsUtil.toModelPermissions(contextCompany.getCompanyId(), permissions, resourceId, resourceName, resourceActionLocalService, resourcePermissionLocalService, roleLocalService));
```
Comment (nhpatt) [inline modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/base_resource_impl.ftl:219]: Can a macro be extracted here?

File: modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/base_resource_test_case.ftl [hunk 2]
```diff
@@ -1139,6 +1139,121 @@ public abstract class Base${schemaName}ResourceTestCase {
...
+
+				com.liferay.portal.kernel.model.Role role = RoleTestUtil.addRole(RoleConstants.TYPE_REGULAR);
+
+				assertHttpResponseStatusCode(
+				200,
+				${schemaVarName}Resource.${javaMethodSignature.methodName}HttpResponse(
+					<#list javaMethodSignature.javaMethodParameters as javaMethodParameter>
+						<#if freeMarkerTool.isPathParameter(javaMethodParameter, javaMethodSignature.operation) && stringUtil.equals(javaMethodParameter.parameterName, schemaVarName + "Id")>
+							${schemaVarName}.getId()
+						<#elseif freeMarkerTool.isPathParameter(javaMethodParameter, javaMethodSignature.operation) && properties?keys?seq_contains(javaMethodParameter.parameterName)>
```
Comment (nhpatt) [inline modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/base_resource_test_case.ftl:1157]: Can we extract this to a macro too? say yes :P

File: modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/base_resource_test_case.ftl [hunk 3]
```diff
@@ -1139,6 +1139,121 @@ public abstract class Base${schemaName}ResourceTestCase {
...
+						<#sep>, </#sep>
+					</#list>
+					));
+
+					assertHttpResponseStatusCode(
+					404,
+					${schemaVarName}Resource.${javaMethodSignature.methodName}HttpResponse(
+					<#list javaMethodSignature.javaMethodParameters as javaMethodParameter>
+						<#if freeMarkerTool.isPathParameter(javaMethodParameter, javaMethodSignature.operation) && stringUtil.equals(javaMethodParameter.parameterName, schemaVarName + "Id")>
+							<#if stringUtil.equals(javaMethodParameter.parameterType, "java.lang.Double")>
```
Comment (nhpatt) [inline modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/base_resource_test_case.ftl:1192]: And this... can be extracted to a macro? 😂

**Comment by nhpatt** [comment]
Ok, so we change a `void` return type to return a `Page<Permission>`. I guess it doesn't change baseline because if someone was using the Java method (and expecting nothing in return) it'll keep working without changes and it doesn't add a method. I would more comfortable at least changing a minor, but if SF does not complain, let it be.

It's the safest change in the API side... it can still break some things due to size or strange code expecting empty return. But I think we are Ok here, it's worth it IMHO.

So go ahead, after CI issues :D

**Comment by javierdearcos** [comment]
Great! I'll extract the macros then to clean the code and send it. Thanks!

Another thing it that we are lying in the OpenAPI saying that we don't return anything in the get and put permissions but I don't know how to handle this... As there is no component in the OpenAPI I don't know how to define it there

**Comment by nhpatt** [comment]
Oh, I see. Because we sneakily return `Permissions`. We could add another convention to the OpenAPI... but we already return `Permission` and `PagePermission` in the real OpenAPI (tm). This is just the fake one.

So we can see this already as a creation convention, I think we are Ok on this, as we are with `Pagination` or `Page` (I would fix this one first). We are definitely not Ok with the convention for `Multipart` though, but there is a ticket for that.

**Comment by javierdearcos** [comment]
ci:test:sf

**Comment by javierdearcos** [comment]
ci:test:relevant

**Comment by javierdearcos** [comment]
ci:forward

**Comment by javierdearcos** [comment]
ci:forward

**Pull request closed without merging**